lib/python/Pyxsim/__init__.py

```diff
@@ -116,10 +155,13 @@ def run_on_simulator(*args, **kwargs):
     return result
 
 
-def do_run_pyxsim(xe, simargs, appargs, simthreads):
+def do_run_pyxsim(xe, simargs, appargs, simthreads, plugins=None):
     xsi = pyxsim.Xsi(xe_path=xe, simargs=simargs, appargs=appargs)
     for x in simthreads:
         xsi.register_simthread(x)
+    if plugins:
+        for plugin in plugins:
+            xsi.register_plugin(plugin)
```

Comment:
I'd just be using plugins via sim args, do we need to support plugin args?

Reply:
So XsiPlugin is a class that's defined, see e.g. pyxsim.py, line 454, where an XsiLoopbackPlugin is defined that seems to replicate the Loopback plugin in xsim. This was the last piece needed to actually use it, although whether it needs to exist at all is a good question.
In /theory/ you could define Pyxsim plugins that are clocked alongside simthreads, although in practice why you would choose to make an XsiPlugin rather than a SimThread escapes me (see e.g. the Clock class in lib_i2s/i2s_master_checker.py) - they seem like they're doing the same thing, but plugins are always clocked rather than simthreads which are waited upon.

Reply:
Ah yes, only just noticed this. I must say not sure I understand the utility of this given this whole framework is essentially for creating plugin functionality..